.github/workflows/analyzers.yaml

```diff
@@ -0,0 +1,95 @@
+name: Run analyzers
+
+on:
+  push:
+    branches: [main]
+  pull_request:
+    branches: [main]
+  workflow_dispatch:
+
+jobs:
+  cppcheck:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Install dependencies
+        run: |
+          sudo apt-get update
+          sudo DEBIAN_FRONTEND=noninteractive apt-get -y install cppcheck
```

Comment:
Do you have any opinion on installing cppcheck via Ubuntu's package manager vs. compiling it from source to get a more recent version? In `si4ti` cppcheck is/was compiled from source and I am wondering if either way is preferable over the other.